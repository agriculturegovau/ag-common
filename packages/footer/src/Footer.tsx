import { useMemo } from 'react';
import { Box } from '@ag.ds-next/react/box';
import { CoreProvider, tokens } from '@ag.ds-next/react/core';
import { Footer as AgDsFooter, FooterDivider } from '@ag.ds-next/react/footer';
import { LinkList } from '@ag.ds-next/react/link-list';
import { Text } from '@ag.ds-next/react/text';

const footerLinks = [
	{
		href: '/about',
		label: 'About',
	},
	{
		href: '/about/contact-us',
		label: 'Contact us',
	},
	{
		href: '/help',
		label: 'Help',
	},
	{
		href: 'https://www.agriculture.gov.au/about/commitment/accessibility',
		rel: 'external',
		label: 'Accessibility',
	},
	{
		href: 'https://www.agriculture.gov.au/about/disclaimer',
		rel: 'external',
		label: 'Disclaimer',
	},
	{
		href: '/privacy',
		label: 'Privacy',
	},
];

export const Footer = () => {
	const year = useMemo(() => new Date().getFullYear(), []);
	return (
		<CoreProvider>
			<Box palette="dark">
				<AgDsFooter background="bodyAlt">
					<nav aria-label="footer">
						<LinkList links={footerLinks} horizontal />
					</nav>
					<FooterDivider />
					<Text fontSize="xs" maxWidth={tokens.maxWidth.bodyText}>
						We acknowledge the traditional owners of country throughout
						Australia and recognise their continuing connection to land, waters
						and culture. We pay our respects to their Elders past, present and
						emerging.
					</Text>
					<Text fontSize="xs" maxWidth={tokens.maxWidth.bodyText}>
						&copy; {year} Department of Agriculture, Fisheries and Forestry
					</Text>
				</AgDsFooter>
			</Box>
		</CoreProvider>
	);
};
