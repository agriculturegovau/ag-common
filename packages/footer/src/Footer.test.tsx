import { Footer } from './Footer';

describe('Footer', () => {
	it('exists', () => {
		expect(Footer).toBeDefined();
	});
});
