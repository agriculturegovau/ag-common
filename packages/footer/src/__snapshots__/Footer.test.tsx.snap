// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<div>
  <div
    class="css-v7su81-dark-dark-boxStyles-boxStyles"
  >
    <footer
      class="css-16rf0on-boxStyles-boxStyles-Footer"
    >
      <div
        class="css-1a1hdgf-boxStyles-boxStyles"
      >
        <nav
          aria-label="footer"
        >
          <ul
            class="css-q7n2pk-boxStyles-boxStyles"
          >
            <li
              class="css-1b13c5y-boxStyles-boxStyles"
            >
              <a
                class="css-1kl9y73-TextLink-TextLink"
                href="/about"
              >
                About
              </a>
            </li>
            <li
              class="css-1b13c5y-boxStyles-boxStyles"
            >
              <a
                class="css-1kl9y73-TextLink-TextLink"
                href="/about/contact-us"
              >
                Contact us
              </a>
            </li>
            <li
              class="css-1b13c5y-boxStyles-boxStyles"
            >
              <a
                class="css-1kl9y73-TextLink-TextLink"
                href="/help"
              >
                Help
              </a>
            </li>
            <li
              class="css-1b13c5y-boxStyles-boxStyles"
            >
              <a
                class="css-1kl9y73-TextLink-TextLink"
                href="https://www.agriculture.gov.au/about/commitment/accessibility"
                rel="external"
              >
                Accessibility
              </a>
            </li>
            <li
              class="css-1b13c5y-boxStyles-boxStyles"
            >
              <a
                class="css-1kl9y73-TextLink-TextLink"
                href="https://www.agriculture.gov.au/about/disclaimer"
                rel="external"
              >
                Disclaimer
              </a>
            </li>
            <li
              class="css-1b13c5y-boxStyles-boxStyles"
            >
              <a
                class="css-1kl9y73-TextLink-TextLink"
                href="/privacy"
              >
                Privacy
              </a>
            </li>
          </ul>
        </nav>
        <hr
          aria-hidden="true"
          class="css-23xxsd-boxStyles-boxStyles-FooterDivider"
        />
        <span
          class="css-a4lzj2-boxStyles-boxStyles"
        >
          We acknowledge the traditional owners of country throughout Australia and recognise their continuing connection to land, waters and culture. We pay our respects to their Elders past, present and emerging.
        </span>
        <span
          class="css-a4lzj2-boxStyles-boxStyles"
        >
          Â© 
          2024
           Department of Agriculture, Fisheries and Forestry
        </span>
      </div>
    </footer>
  </div>
</div>
`;
